src/redux/slices/user.js

```diff
@@ -213,27 +216,107 @@ export const transferNFT = createAsyncThunk(
   }
 );
 
+export const getContacts = createAsyncThunk(
+  'keyring/getContacts',
+  async (walletNumber = 0, { getState, dispatch }) => {
+    try {
+      dispatch(setContactsLoading(true));
+      const state = getState();
+      const res = await state.keyring.instance?.getContacts(walletNumber);
+      return res?.map(formatContact);
+    } catch (e) {
+      dispatch(setContactsLoading(false));
+      console.log('Error getting contacts:', e);
+    }
+  }
+);
+
+export const addContact = createAsyncThunk(
+  'keyring/addContact',
+  async ({ contact, walletNumber = 0, onFinish }, { getState, dispatch }) => {
+    try {
+      dispatch(setContactsLoading(true));
+      const state = getState();
+      const res = await state.keyring.instance?.addContact(
+        formatContactForController(contact),
+        walletNumber
+      );
+      if (res) {
+        dispatch(getContacts())
```

Comment:
what if instead of calling getContacts again, we concat the new contact in the current list? To avoid that request

Reply:
We could do that, but we need to make this request since we need to update the contact list, because you can interact with it in mobile and also in extension

Reply:
Yeah, i know, but will that be a common case? A user changing contacts at the same time in both platforms?